eras/byron/ledger/impl/test/Test/Cardano/Chain/Slotting/Properties.hs

```diff
@@ -66,16 +65,10 @@ ts_prop_addSlotCount = withTestsTS 100 . property $ do
   sc <- forAll genSlotCount
   fs <- forAll genSlotNumber
   let added = fs + (SlotNumber $ unSlotCount sc)
-  addSlotCount sc fs === added
-
--- Check that `subSlotCount` actually subtracts.
-ts_prop_subSlotCount :: TSProperty
-ts_prop_subSlotCount = withTestsTS 100 . property $ do
-  sc <- forAll genSlotCount
-  fs <- forAll genSlotNumber
-  let sc' = SlotNumber $ unSlotCount sc
-      subtracted = fs - sc'
-  subSlotCount sc fs === if fs > sc' then subtracted else SlotNumber 0
+  addSlotCount sc fs
+    === if unSlotNumber fs <= maxBound - (unSlotCount sc)
+      then added
+      else SlotNumber maxBound
```

Comment:
this test feels pretty silly, it's the same logic as the function it is testing... and in general I feel like `addSlotCount` is only a bit less of a footgun than `subSlotCount` (the ordering on `subSlotCount` was pretty wacky).